### PR TITLE
Conditional Go Delve patch

### PR DESCRIPTION
**Pull request by Razz4780**
Issue #284

### CODE REVIEW

File: modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt [hunk 1]
```diff
@@ -75,35 +81,89 @@ class GolandRunConfigurationExtension : GoRunConfigurationExtension() {
...
+        }
+
+        if (!delveExecutable.canExecute()) {
+            delveExecutable.setExecutable(true)
+        }
+
+        val ourDelveVersion = try {
+            getDelveVersion(ourDelvePath)
+        } catch (e: Throwable) {
+            MirrordLogger.logger.error("Failed to get version delve bundled with mirrord plugin ($ourDelvePath)", e)
```
Comment (DmitryDodzin) [inline modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt:104]: Is there a possobility to enter this catch from a "correct" flow? like should there be any reason for the version to fail on our shipped dlv other then somthing unexpected (unsupported system ect...)

File: modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt [hunk 2]
```diff
@@ -75,35 +81,89 @@ class GolandRunConfigurationExtension : GoRunConfigurationExtension() {
...
+
+        // parameters returns a copy, and we can't modify it so need to reset it.
+        val patchedParameters = executor.parameters
+        for (i in 0 until patchedParameters.size) {
+            // no way to reset the whole commandline, so we just remove each entry.
+            executor.withoutParameter(0)
+
+            val foundDelvePath = patchedParameters[i].toPresentableString()
+
+            if (!foundDelvePath.endsWith("/dlv", true)) {
```
Comment (DmitryDodzin) [inline modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt:117]: Is is feesable that there will be a relative path here? like a singular string of `dlv`?

**Review by DmitryDodzin — COMMENTED** [review]
Looks fine I just have a couple of questions

File: modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt (hunk 1 above)
Comment (Razz4780) [reply modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt:104]: Maybe some weird remote execution contexts, but I'm not sure. In most scenarios, this failure would mean that our build logic is buggy. Do you think an exception would be better?

File: modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt (hunk 2 above)
Comment (Razz4780) [reply modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt:117]: Not sure, I left this check as it was. Have been working fine so far :sweat_smile:

File: modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt (hunk 1 above)
Comment (Razz4780) [reply modules/products/goland/src/main/kotlin/com/metalbear/mirrord/products/goland/GolandRunConfigurationExtension.kt:108]: Fixed

**Review by DmitryDodzin — APPROVED** [review]

**Pull request merged**